packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -265,24 +282,37 @@ class CupertinoSliverNavigationBar extends StatelessWidget {
 
 /// Returns `child` wrapped with background and a bottom border if background color
 /// is opaque. Otherwise, also blur with [BackdropFilter].
-Widget _wrapWithBackground({Color backgroundColor, Widget child}) {
+Widget _wrapWithBackground({
+  Color borderColor,
+  Color backgroundColor,
+  bool adjustSystemUiOverlayStyle,
+  Widget child,
+}) {
+  BoxBorder border;
+
+  if (borderColor != null && borderColor.alpha > 0) {
```

Comment:
we should only do this for null; changing the alpha alone should not change the layout.